plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/project/ManageProjectsWizardPage.java

```diff
@@ -0,0 +1,291 @@
+
+	public ManageProjectsWizardPage(String title, String description, ManageProjectsWizardPageModel pageModel, IWizard wizard) {
+		super(title, description, title, wizard);
+		this.pageModel = pageModel;
+	}
+
+	protected void doCreateControls(Composite parent, DataBindingContext dbc) {
+		GridLayoutFactory.fillDefaults().margins(10, 10).applyTo(parent);
+
+		Group group = new Group(parent, SWT.NONE);
+		group.setText("OpenShift Projects");
+		GridDataFactory.fillDefaults()
+				.align(SWT.FILL, SWT.FILL).grab(true, true).applyTo(group);
+		GridLayoutFactory.fillDefaults()
+				.numColumns(2).margins(6, 6).applyTo(group);
+
+		// table
+		Composite tableContainer = new Composite(group, SWT.NONE);
+		this.viewer = createTable(tableContainer);
+		GridDataFactory.fillDefaults()
+				.span(1, 5).align(SWT.FILL, SWT.FILL).grab(true, true).applyTo(tableContainer);
+		viewer.setContentProvider(new ObservableListContentProvider());
+		viewer.setInput(BeanProperties.list(
+				ManageProjectsWizardPageModel.PROPERTY_PROJECTS).observe(pageModel));
+		loadProjects(dbc);
+		IObservableValue viewerSingleSelection = ViewerProperties.singleSelection().observe(viewer);
+		ValueBindingBuilder.bind(viewerSingleSelection)
+				.to(BeanProperties.value(ManageProjectsWizardPageModel.PROPERTY_SELECTED_PROJECT).observe(pageModel))
+				.in(dbc);
+
+		// new 
+		Button newButton = new Button(group, SWT.PUSH);
+		GridDataFactory.fillDefaults()
+				.align(SWT.FILL, SWT.FILL).applyTo(newButton);
+		newButton.setText("New...");
+		newButton.addSelectionListener(onNew());
+
+		// remove
+		Button removeButton = new Button(group, SWT.PUSH);
+		GridDataFactory.fillDefaults()
+				.align(SWT.FILL, SWT.FILL).applyTo(removeButton);
+		removeButton.setText("Remove...");
+		removeButton.addSelectionListener(onRemove(dbc));
+		ValueBindingBuilder
+				.bind(WidgetProperties.enabled().observe(removeButton))
+				.notUpdatingParticipant()
+				.to(viewerSingleSelection)
+				.converting(new IsNotNull2BooleanConverter())
+				.in(dbc);
+
+		Composite filler = new Composite(group, SWT.None);
+		GridDataFactory.fillDefaults()
+				.align(SWT.FILL, SWT.FILL).applyTo(filler);
+
+		// refresh
+		Button refreshButton = new Button(group, SWT.PUSH);
+		GridDataFactory.fillDefaults()
+				.align(SWT.FILL, SWT.END).applyTo(refreshButton);
+		refreshButton.setText("Refresh...");
+		refreshButton.addSelectionListener(onRefresh(dbc));
+	}
+
+	private void loadProjects(DataBindingContext dbc) {
+		try {
+			org.jboss.tools.common.ui.WizardUtils.runInWizard(
+					new AbstractDelegatingMonitorJob("Loading OpenShift projects...") {
+
+						@Override
+						protected IStatus doRun(IProgressMonitor monitor) {
+							pageModel.loadProjects();
+							return Status.OK_STATUS;
+						}
+					}, new DelegatingProgressMonitor(), getContainer(), dbc);
+		} catch (InvocationTargetException e) {
+			LOG.logError(NLS.bind("Could not load projects for connection {0}", pageModel.getConnection().toString()), e);
```

Comment:
this looks like a case for the multi-catch close in jdk7 causing this block to get simpler:
} catch (InvocationTargetException | InterruptedException e)  {
      LOG.logError(NLS.bind("Could not load projects for connection {0}", pageModel.getConnection().toString()), e);
}